great_expectations/execution_engine/redshift_execution_engine.py

```diff
@@ -82,43 +82,30 @@ def _sqlalchemy(
         # For sqlalchemy 2 use this new implementation which avoids incompatible parts of dialect
         # Get the table information
         assert isinstance(execution_engine, RedshiftExecutionEngine)
+        result = []
         table_name, schema_name = cls._get_table_schema(execution_engine, metric_domain_kwargs)
-        full_table_name = f'"{schema_name}"."{table_name}"' if schema_name else f"{table_name}"
-        # Query for the column information
-        query = sa.text(f"""
-            select * from pg_get_cols('{full_table_name}')
-            cols(view_schema name, view_name name, col_name name, col_type varchar, col_num int);
-        """)
-        raw_result = execution_engine.execute_query(query)
-        # Parse out metadata
-        column_metadata = []
-        # The raw result is a tuple of strings, one for each row.
-        for r in raw_result:
-            _schema, _table, column, raw_column_type, _column_num = r
-            # If the type is parameterized, we must removed the arguments to the types to do a
-            # string lookup from the type string to the actual type.
-            column_type_str_parts = raw_column_type.split("(")
-            column_base_type = REDSHIFT_TYPES.get(column_type_str_parts[0])
-            if column_base_type is None:
+        schema_filter: str = f"and table_schema = '{schema_name}'" if schema_name else ""
+        query = sa.text(
+            f"""
+            SELECT column_name, data_type, character_maximum_length
```

Comment:
It looks like `character_maximum_length` is for some but not all data types according to the docs. It looks like you want to use numeric_precision or datetime_precision should be used for other datatypes. It seems like all but 1 of those should be null. You can also query for numeric_scale to get the second argument if it exists.
If you read those in and make an `args` list that you append the non null values you, you can pass in `*args` to `redshift_type(*args)` below.